include/flang/semantics/symbol.h

```diff
@@ -55,6 +55,10 @@ class SubprogramDetails {
   SubprogramDetails() {}
   SubprogramDetails(const SubprogramDetails &that)
     : dummyArgs_{that.dummyArgs_}, result_{that.result_} {}
+  SubprogramDetails &operator=(const SubprogramDetails &) = delete;
+  ~SubprogramDetails() = default;
```

Comment:
Was this explicitly defaulted destructor necessary?

Reply:
It was not, thanks. I have removed all added constructor/destructor and the copy constructor after Tim's comment.